libs/common-system/src/main/java/org/codice/ddf/configuration/SystemBaseUrl.java

```diff
@@ -0,0 +1,217 @@
+     *
+     * @param proto   Protocol to use during url construction. A null value will
+     *                cause the system default protocol to be used
+     * @param context The context path to be appened to the end of the base url
+     * @return
+     */
+    public String constructUrl(String proto, String context) {
+        return constructUrl(proto, context, false);
+    }
+
+    /**
+     * Construct a url based on the protocol and context
+     *
+     * @param proto    Protocol to use during url construction. A null value will
+     *                 cause the system default protocol to be used
+     * @param context  The context path to be appened to the end of the base url
+     * @param external If true will return the external url
+     * @return
+     */
+    public String constructUrl(String proto, String context, boolean external) {
+        StringBuilder sb = new StringBuilder();
+        if (proto == null) {
+            sb.append(getProtocol());
+        } else {
+            sb.append(proto);
+        }
+        if (!sb.toString().endsWith("://")) {
+            sb.append("://");
+        }
+        sb.append(external ? getExternalHost() : getHost());
+        sb.append(":");
+        sb.append(getPort(proto, external));
+
+        if (context != null) {
+            if (!context.startsWith("/")) {
+                sb.append("/");
+            }
+            sb.append(context);
+        }
+
+        return sb.toString();
+    }
+
+    /**
+     * Gets the port number for the given protocol. If the protocol is null or not one of
+     * http or https returns the https port number.
+     *
+     * @param proto    protocol (http or https)
+     * @param external return external port
+     * @return
+     */
+    public String getPort(String proto, boolean external) {
+        if (proto != null && !proto.startsWith("https")) {
+            return external ? getExternalHttpPort() : getHttpPort();
+        } else {
+            return external ? getExternalHttpsPort() : getHttpsPort();
+        }
+    }
+
+    public String getExternalHost() {
+        return getProperty(EXT_HOST, DEFAULT_EXT_HOST);
+    }
+
+    public String getHttpPort() {
+        return getProperty(HTTP_PORT, DEFAULT_EXT_HTTP_PORT);
+    }
+
+    public String getHttpsPort() {
+        return getProperty(HTTPS_PORT, DEFAULT_HTTPS_PORT);
+    }
+
+    public String getHost() {
+        return getProperty(HOST, DEFAULT_HOST);
+    }
+
+    public String getProtocol() {
+        return getProperty(PROTOCOL, DEFAULT_PROTOCOL);
+    }
+
+    public String getExternalHttpPort() {
+        return getProperty(EXT_HTTP_PORT, DEFAULT_EXT_HTTP_PORT);
+    }
+
+    public String getExternalHttpsPort() {
+        return getProperty(EXT_HTTPS_PORT, DEFAULT_EXT_HTTPS_PORT);
+    }
+
+    public String getRootContext() {
+        return getProperty(ROOT_CONTEXT, DEFAULT_ROOT_CONTEXT);
+    }
+
+    private String getProperty(String propName, String defaultValue) {
```

Comment:
System already has this method.

http://docs.oracle.com/javase/7/docs/api/java/lang/System.html#getProperty(java.lang.String,%20java.lang.String)